code-challenges.test.js

```diff
@@ -10,33 +10,152 @@
 const secretCodeWord1 = "Lackadaisical"
 // Expected output: "L4ck4d41s1c4l"
 const secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
+// a) Create a test with expect statements using the variables provided.
+//Pseudo code
+// Create a describe statement 
+
+
+describe ("messageCoder", () => {
+    it("takes in a function and returns a string with a coded message. The coded message converts a to 4, e to 3, i to 1, and o to 0.",  () =>{
+        const secrectCodeWord1 = "Lackadaisical"
+        const secretCodeWord2 = "Gobbledygook"
+        const secretCodeWord3 = "Eccentric"
+         
+        expect(messageCoder(secrectCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(messageCoder(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(messageCoder(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+// FAIL  ./code-challenges.test.js
+// ● Test suite failed to run
+
+//   SyntaxError: /Users/learnacademy/Desktop/week-5-assessment-ValC001/code-challenges.test.js: Unexpected token, expected "," (24:99)
+
+   // 22 | // Create a describe statement 
+//     23 | describe ("messageCoder", () => {
+//   > 24 |     it("takes in a function and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.",  () =>{
+//        |                                                                                                    ^
+//     25 |         const secrectCodeWord1 = "Lackadaisical"
+//     26 |         const secretCodeWord2 = "Gobbledygook"
+//     27 |         const secretCodeWord3 = "Eccentric"
+
+//     Test Suites: 1 failed, 1 total
+//     Tests:       0 total
+//     Snapshots:   0 total
+//     Time:        0.875 s
+//     Ran all test suites matching /code-challenges.test.js/i.
+//     error Command failed with exit code 1.
+
 
 // b) Create the function that makes the test pass.
 
-// --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
-// a) Create a test with expects statement using the variable provided.
+//Psuedo Code
+//I can use replace method to replace letters with numbers. I can do an if else method to detect if it has certain letter and if it does replace it.
+//The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
+// const secrectCodeWord1 = "Lackadaisical"
+// const secretCodeWord2 = "Gobbledygook"
+// const secretCodeWord3 = "Eccentric"
+ 
+
+function messageCoder(secrectCodeWord1){
+    return secrectCodeWord1
+    .replaceAll('a','4')
+    .replaceAll('A','4')
+    .replaceAll('e','3')
+    .replaceAll('E','3')
+    .replaceAll('i','1')
+    .replaceAll('I','1')
+    .replaceAll('o','0')
+    .replaceAll('o','0')
+}
```

Comment:
I see you found the `.replaceAll` method, nice job! I suggest adding that into your pseudo code as well